docs/DeveloperGuide.md

```diff
@@ -346,42 +347,420 @@ Priorities: High (must have) - `* * *`, Medium (nice to have)  - `* *`,  Low (un
+|`*`|User|add a message of the day|customise the application|
 
-{More to be added}
 
 ## Appendix B : Use Cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is the `TaskManager`)
+
+| Use Case | Description |
+|-----|-----|
+|[Display help](#use-case-display-help)|Displays a list of possible instructions|
+|[Add task](#use-case-add-task)|Adds a task to the todo list. The task may have a concrete or floating time. Including tags at the end of the task is optional|
+|[Delete task](#use-case-delete-task)|Removes a task from the todo list|
+|[Mark task as complete](#use-case-complete-task)|Marks a task done by its name|
+|[List Upcoming Deadlines](#use-case-list-upcoming-deadlines)|Lists the tasks with upcoming deadlines within the next 2 weeks|
+|[List uncompleted tasks](#use-case-list-uncompleted-tasks)| Lists all uncompleted tasks|
+|[List completed tasks](#use-case-list-completed-tasks)| Lists all completed tasks|
+|[Calendar View](#use-case-calendar-view)|View the todo list as a calendar|
+|[Sync with Google Calendar](#use-case-sync-with-google-calendar)|Synchronise the todo list with a Google Calendar|
+|[Customize File Storing](#use-case-customize-file-storing)|Set a custom location for where to save the data|
+|[Reminder mode](#use-case-reminder-mode)|Sets whether the application will send a reminder as a task's deadline draws near|
+|[Attach Links](#use-case-attach-links)|Attaches links to a task|
+|[Show Progress Report](#use-case-show-task-progress-report)|Displays the current statistics on how many tasks have been added, completed, and pending completion|
+|[Search](#use-case-search)|Displays a list of tasks with matching keywords as queried|
+|[View Overdue Tasks](#use-case-view-overdue-tasks)|Displays a list of tasks that are overdue|
+|[Undo Previous Command](#use-case-undo-previous-command)|Undo the previous command|
+|[Repeat Task](#use-case-repeat-task)|Repeat the task every week|
+|[Add Tags to Created Task](#use-case-add-tags-to-created-task)|Add one or more tags to a task to label it|
+|[Customise Message of the Day](#use-case-customise-message-of-the-day)|Add a message of the day that is displayed each time the program starts up|
+|[Scroll Through Previous Commands](#use-case-scroll-through-previous-commands)|Iterate through previously executed commands|
 
-#### Use case: Delete person
+---
 
-**MSS**
+### Use Case: Display Help
 
-1. User requests to list persons
-2. AddressBook shows a list of persons
-3. User requests to delete a specific person in the list
-4. AddressBook deletes the person <br>
-Use case ends.
+#### Main Success Scenario:
 
-**Extensions**
+1. User requests to display help.
+2. System displays the help messages.
+3. Use case ends.
 
-2a. The list is empty
+##### Extensions
 
-> Use case ends
+1a. User entered an invalid command.
```

Comment:
This can be considered under another use case. Maybe something like "Use Case: Enter Command". You can assume the Step num 1 of each command use case to then be already valid